File: src/main/java/com/hileco/cortex/instructions/conditions/EQUALS.java
package com.hileco.cortex.instructions.conditions;

import lombok.EqualsAndHashCode;

import java.util.Arrays;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
public class EQUALS extends ConditionInstruction {
    @Override
    public boolean innerExecute(byte[] left, byte[] right) {
        return Arrays.equals(left, right);
    }
}

File: src/main/java/com/hileco/cortex/instructions/conditions/EQUALS.kt
package com.hileco.cortex.instructions.conditions

import com.hileco.cortex.constraints.expressions.Expression
import com.hileco.cortex.constraints.expressions.Expression.*
import java.math.BigInteger

class EQUALS : ConditionInstruction() {
    override fun innerExecute(left: ByteArray, right: ByteArray): Boolean {
        return BigInteger(left) == BigInteger(right)
    }

    override fun innerExecute(left: Expression, right: Expression): Expression {
        if (left is Value && right is Value) {
            return if (left.constant == right.constant) True else False
        }
        return Equals(left, right)
    }
}
